src/octoprint/plugins/gcodeviewer/static/js/viewer/reader.js

```diff
@@ -210,35 +210,18 @@ GCODE.gCodeReader = (function () {
         loadFile: function (reader) {
             this.clear();
 
-            var totalSize = reader.target.result.length;
-
-            /*
-             * Split by line ending
-             *
-             * Be aware that for windows line endings \r\n this leaves the \r attached to
-             * the lines. That will not influence our parser, but makes file position
-             * calculation way easier (line length + 1), so we just leave it in.
-             *
-             * This cannot cope with old MacOS \r line endings, but those should
-             * really not be used anymore and thus we'll happily ignore them here.
-             *
-             * Note: A simple string split uses up *much* less memory than regex.
-             */
-            lines = reader.target.result.split("\n");
-
-            reader.target.result = null;
-            prepareGCode(totalSize);
-
             var mustCompress =
                 gCodeOptions["forceCompression"] ||
                 gCodeOptions["alwaysCompress"] ||
                 (gCodeOptions["compressionSizeThreshold"] > 0 &&
                     gCodeOptions["compressionSizeThreshold"] <= totalSize);
 
             GCODE.ui.worker.postMessage({
-                cmd: "parseGCode",
+                cmd: "downloadAndParseGCode",
                 msg: {
-                    gcode: gcode,
+                    path: reader.path,
+                    localPath: reader.localPath,
```

Comment:
`path` vs `localPath` confused me a bit during review further down. How about `url` and `path` instead?